app/controllers/identities/omniauth_callbacks_controller.rb

```diff
@@ -30,5 +30,19 @@ def shibboleth
       redirect_to new_identity_registration_url
     end
   end
-end
 
+  def cas
+    @identity = Identity.find_for_cas_oauth(request.env["omniauth.auth"], current_identity)
+
+    if @identity.persisted?
+      sign_in_and_redirect @identity, :event => :authentication
```

Comment:
We're working on better styling our Ruby code. Reference - https://github.com/bbatsov/ruby-style-guide

Here, use new Ruby syntax instead of hash rockets, e.g.

     event: :authentication

Parentheses for method arguments

     sign_in_and_redirect(@identity ...)